test/e2e/external_auth_complete_cluster.go

```diff
@@ -0,0 +1,242 @@
+	var sf entraSecretFileStep2
+	Expect(json.Unmarshal(b, &sf)).To(Succeed(), "failed to parse Entra secret JSON at %s", path)
+	Expect(sf.TenantID).NotTo(BeEmpty())
+	Expect(sf.ClientID).NotTo(BeEmpty())
+	return sf
+}
+
+// ---------- helpers ----------
+func majorMinor(v string) string { // normalize "4.19.0" -> "4.19"
+	p := strings.SplitN(v, ".", 3)
+	if len(p) >= 2 {
+		return p[0] + "." + p[1]
+	}
+	return v
+}
+
+func applyVersionToTemplate(tpl []byte, version string) []byte { // replace placeholder/patch value
+	v := majorMinor(version)
+	s := string(tpl)
+	s = strings.ReplaceAll(s, "VERSION_REPLACE_ME", v)
+	// safety if a concrete 4.x.y slipped in:
+	s = strings.ReplaceAll(s, `"id": "4.19.0"`, `"id": "`+v+`"`)
+	return []byte(s)
+}
+
+func httpClientInsecure() *http.Client {
+	insecure := true
+	if v := os.Getenv("INSECURE_SKIP_TLS"); strings.TrimSpace(strings.ToLower(v)) == "false" {
+		insecure = false
+	}
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: insecure}, // dev/self-signed
+	}
+	return &http.Client{Transport: tr, Timeout: 45 * time.Second}
+}
+
+var _ = Describe("ExternalAuth Full E2E", func() {
+	It("creates a full HCP cluster and applies ExternalAuth config",
+		labels.RequireNothing, labels.Critical, labels.Positive, labels.ExternalEntra,
+		func(ctx context.Context) {
+			tc := framework.NewTestContext()
+
+			const (
+				region                 = "uksouth"
+				customerNSGName        = "customer-nsg-name"
+				customerVnetName       = "customer-vnet-name"
+				customerVnetSubnetName = "customer-vnet-subnet1"
+				customerClusterName    = "external-auth-cluster"
+				customerNodePoolName   = "np-1"
+
+				// ExternalAuth constants (non-secret)
+				externalAuthID            = "e2e-hypershift-oidc"
+				usernameClaim             = "email"
+				groupsClaim               = "groups"
+				externalAuthComponentName = "console"
+				externalAuthComponentNS   = "openshift-console"
+			)
+
+			By("creating resource group")
+			resourceGroup, err := tc.NewResourceGroup(ctx, "external-auth", region)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("provisioning infra")
+			_, err = framework.CreateBicepTemplateAndWait(ctx,
+				tc.GetARMResourcesClientFactoryOrDie(ctx).NewDeploymentsClient(),
+				*resourceGroup.Name, "infra",
+				framework.Must(TestArtifactsFS.ReadFile("test-artifacts/generated-test-artifacts/standard-cluster-create/customer-infra.json")),
+				map[string]interface{}{
+					"customerNsgName":        customerNSGName,
+					"customerVnetName":       customerVnetName,
+					"customerVnetSubnetName": customerVnetSubnetName,
+				}, 45*time.Minute)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("creating HCP cluster (injecting MAJOR.MINOR version)")
+			rawClusterTpl := framework.Must(
+				TestArtifactsFS.ReadFile("test-artifacts/generated-test-artifacts/standard-cluster-create/cluster.json"),
+			)
+			ver := strings.TrimSpace(os.Getenv("HCP_OPENSHIFT_VERSION"))
```

Comment:
Not an env var.  Hard code for now, we can make a loop and run one for every version we support later.